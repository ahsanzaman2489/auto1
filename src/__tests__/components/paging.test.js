import React from 'react';
import PagingComponent from '../../components/paging';
import {shallow} from 'enzyme';


const setup = (props = {}, state = null) => {
    const defaultProps = {totalPageCount: 20, location: {search: "?color=black&manufacturer=Chrysler&page=1&sort=asc"}};
    const setUpProps = {...defaultProps, ...props};
    const wrapper = shallow(<PagingComponent {...setUpProps}/>);
    if (state) wrapper.setState(state);
    return {wrapper, props: setUpProps}
};


describe("Paging component", () => {
    it('should render 5 list items', () => {
        const {wrapper} = setup();
        expect(wrapper.find('li').length).toBe(5);
    });

    it('first two list have class disabled when current page is one', () => {
        const {wrapper} = setup();

        expect(wrapper.find('ul').childAt(0).find('NavLink').props().className).toBe("disabled");
        expect(wrapper.find('ul').childAt(1).find('NavLink').props().className).toBe("disabled");
    });

    it('third list should have text of total and current page', () => {
        const {wrapper,props} = setup();

        expect(wrapper.find('ul').childAt(2).text()).toBe(`page 1 of ${props.totalPageCount}`);
    });

    it('last two list have class disabled when current page is equal to total page count', () => {
        const {wrapper} = setup({location: {search: "?color=black&manufacturer=Chrysler&page=20"}});

        expect(wrapper.find('ul').childAt(3).find('NavLink').props().className).toBe("disabled");
        expect(wrapper.find('ul').childAt(4).find('NavLink').props().className).toBe("disabled");
    });

});


