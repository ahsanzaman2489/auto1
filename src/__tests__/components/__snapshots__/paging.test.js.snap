// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paging component should match with snapshot 1`] = `
<div
  className="pagination"
>
  <ul>
    <li>
      <a
        aria-current={null}
        className="disabled"
        href="/cars/list?color=black&manufacturer=Chrysler&page=1&sort=asc"
        onClick={[Function]}
        style={undefined}
      >
        first
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        className="disabled"
        href="/cars/list?color=black&manufacturer=Chrysler&page=1&sort=asc"
        onClick={[Function]}
        style={undefined}
      >
        previous
      </a>
    </li>
    <li>
      <span>
        page 
        1
         of 
        20
      </span>
    </li>
    <li>
      <a
        aria-current={null}
        className=""
        href="/cars/list?color=black&manufacturer=Chrysler&page=2&sort=asc"
        onClick={[Function]}
        style={undefined}
      >
        next
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        className=""
        href="/cars/list?color=black&manufacturer=Chrysler&page=20&sort=asc"
        onClick={[Function]}
        style={undefined}
      >
        last
      </a>
    </li>
  </ul>
</div>
`;
